### ISSUE TITLE
PanicFilter trims traces too aggressively

### ISSUE THREAD

**Issue by robfig** [revel/revel#273]
Presently, when the panic filter catches a panic, it has a stack like this:

```
/Users/robfig/gocode/src/github.com/robfig/revel/panic.go:19 (0x493a7)
    com/robfig/revel.handleInvocationPanic: ERROR.Print(err, "\n", string(debug.Stack()))
/Users/robfig/gocode/src/github.com/robfig/revel/panic.go:10 (0x5cba9)
    com/robfig/revel.func.039: handleInvocationPanic(c, err)
/Users/robfig/go/src/pkg/runtime/panic.c:229 (0x15271)
    panic: reflect·call(d->fn, (byte*)d->args, d->siz);
/Users/robfig/gocode/src/github.com/robfig/revel/intercept.go:93 (0x5cb3a)
    com/robfig/revel.func.038: panic(err)
/Users/robfig/go/src/pkg/runtime/panic.c:229 (0x15271)
    panic: reflect·call(d->fn, (byte*)d->args, d->siz);
/Users/robfig/go/src/pkg/runtime/panic.c:487 (0x15a13)
    panicstring: runtime·panic(err);
/Users/robfig/go/src/pkg/runtime/os_darwin.c:462 (0x14968)
    sigpanic: runtime·panicstring("invalid memory address or nil pointer dereference");
/Users/robfig/gocode/src/github.com/coopernurse/gorp/gorp.go:949 (0x19328c)
    com/coopernurse/gorp.(*Transaction).Rollback: return t.tx.Rollback()
/Users/robfig/gocode/src/yext/admin/app/controllers/app.go:38 (0x75906)
    Users.PostSpreadsheet: return c.Redirect(routes.Users.Upload())
/Users/robfig/gocode/src/yext/admin/app/controllers/app.go:0 (0x78ee4)
    (*Users).PostSpreadsheet: ???
/Users/robfig/go/src/pkg/reflect/value.go:474 (0x2ffaf)
    Value.call: call(fn, unsafe.Pointer(ptr), uint32(size))
/Users/robfig/go/src/pkg/reflect/value.go:345 (0x2f00b)
    Value.Call: return v.call("Call", in)
/Users/robfig/gocode/src/github.com/robfig/revel/invoker.go:36 (0x48f7f)
    com/robfig/revel.ActionInvoker: resultValue = methodValue.Call(methodArgs)[0]

 ... 15 frames of filters/etc ...
```

It tries to remove the extra stack frames at the beginning, by looking for the first frame that contains the app's path. 

That strategy will strip relevant stack frames if the panic originates in code that lies outside the app path.  In the example, the panic started in gorp due to an incorrectly initialized transaction variable, but Revel stripped it off. 

Alternatives: 
- Strip leading frames that are within Revel 
- Strip leading frames within Revel and standard library
- Strip leading frames within Revel and the runtime package
- Print the entire stack

I think the third option may be the best..

**Comment by wildattire** [comment]
We are repeatedly bitten by this, as will be anyone who breaks their project up into multiple modules.
Our solution is to patch errors.go by replacing BasePath with the GOPATH for any new revel installation.

**Comment by brendensoares** [comment]
@wildattire do you have a PR that we could consider?

**Comment by brendensoares** [comment]
### TODO
- [ ] reproduce this issue and remove truncation (perhaps this should be configurable)

**Comment by jchavannes** [comment]
Created PR with a change that fixes it for me: #1082.

**Comment by notzippy** [comment]
Merged closing
